libnopegl/src/backends/gl/gpu_ctx_gl.c

```diff
@@ -984,6 +984,39 @@ static int gl_get_preferred_depth_stencil_format(struct gpu_ctx *s)
     return NGLI_FORMAT_D24_UNORM_S8_UINT;
 }
 
+static void gl_set_pipeline(struct gpu_ctx *s, struct pipeline *pipeline)
+{
+    struct gpu_ctx_gl *s_priv = (struct gpu_ctx_gl *)s;
+    s_priv->current_pipeline = pipeline;
+}
```

Comment:
1. ngli_gpu_ctx_set_pipeline() currently only store the pipeline, so it would make sense to that, however, I will implement the set_pipeline() mechanism differently in the upcoming PRs and it will be backend specific, for example on Vulkan, set_pipeline() will directly call vkCmdBindPipeline().
2. `pipeline_compat` is an user of the graphics abstraction (not the other way around). Note that there is more change to come to the ngli_gpu_ctx_{draw,draw_indexed,compute}() implementation, ngli_pipeline_*_{draw,draw_indexed,compute} will disappear in the upcoming PRs.